### DISCUSSION TITLE
Communications of Roadmap discussion

### DISCUSSION THREAD

**Discussion by dennyglee** [unitycatalog/unitycatalog#803] [Q&A]
To help clarify the various issues,  I'm taking an [existing Slack thread](https://unitycatalog.slack.com/archives/C0772HF0F7X/p1733853537932379) and copying the relevant components to help with the discussion.  Note, that this initially started with the thread announcing the UC 0.2.1 release:

> From @tdas at 2024-12-10 09:58:57 AM to [general](https://unitycatalog.slack.com/archives/C0772HF0F7X/p1733853537932379)

Hey all! We are excited to announce Unity Catalog 0.2.1 release which has a few bug fixes and a new Python client - https://pypi.org/project/unitycatalog-client/
Artifacts in this release:
- Maven artifacts: [unitycatalog-server](https://central.sonatype.com/artifact/io.unitycatalog/unitycatalog-server/versions), [unitycatalog-client](https://central.sonatype.com/artifact/io.unitycatalog/unitycatalog-client/versions), [unitycatalog-spark_2.12](https://central.sonatype.com/artifact/io.unitycatalog/unitycatalog-spark_2.12/versions), [unitycatalog-spark_2.13](https://central.sonatype.com/artifact/io.unitycatalog/unitycatalog-spark_2.13/versions/)
- PyPI artifacts: https://pypi.org/project/unitycatalog-client/

Key improvements are as follows:
- New Python client - https://pypi.org/project/unitycatalog-client/
  - Note: This will replace the previously released python library - https://pypi.org/project/unitycatalog/. Thanks to our friends at [Daft](https://www.getdaft.io/) for contributing their python client to this core repository so that we (the UC community) can collectively maintain it.
- Tables - Support for MODIFY permission to allow non-owners to write to tables
- Models - API and UI improvements
   - Fixes to API specs
- UI support for creating model, updating model / model version descriptions, and deleting model / model versions,
   - Iceberg catalog APIs - introduced warehouse configuration property for better compatibility with Iceberg connectors
- UI - Improvements
   - Improved cookie handling and support for logging in and out. UI sessions are active for 5 days or whatever timeframe is specified in server.properties.
   - Better security through the use of HttpOnly cookie to manage the session. The session cookie also provides the ability to log out of the UI.
 
> From @ni-mi at 2024-12-10 10:30:03 AM

Can you please get me an answer to my question about v.0.3.0 (that was suppoused to be out this month - which doesn't seem to me like it can happen, although there is no discussion about it that I'm aware of)?
https://unitycatalog.slack.com/archives/C076YREKX8W/p1732708127302739

> From @tdas at 2024-12-10 11:21:26 AM

Unfortunately, we are actually running behind managed tables. :disappointed: In the current larger environment, we are rather investing heavily on the AI library integrations and we are hoping we can get that out, and then resume the managed table effort.

**Comment by dennyglee** [comment]
> From @ognis1205 at 2024-12-10 07:38:03 PM

[@Nimrod Ofek](https://unitycatalog.slack.com/team/U079ZQVGZFA)
 For the same reason, I think this community is not very active.

**Reply by dennyglee** [reply]
Hey @ognis1205 ([@OKAWA Shingo](https://unitycatalog.slack.com/team/U078RM1RN8Y)) - long time no see!  Glad to have you back and you know I always appreciate your candor and contributions!  Saying this, I humbly disagree about the activity.  The unitycatalog-ai library is extremely active and they are doing a lot of updates to the python SDK, AI library integrations, etc.

Could we communicate better, I'm sure we all could do a better job here but note we had weekly/bi-weekly UC community meetups right up until last week.  And with US Thanksgiving and all the winter holidays, I would expect things to be comparatively quieter.   We will kick up bi-weekly UC meetups in the new year and just like we had noted before, this a great forum for us to chat and discuss.  Saying this:
It was recently suggested to me that we should have taken down notes from these meetups and publish them to the UC GitHub discussions so that way decisions and discussion points could be provided without needing to attend or re-watch the videos.  IMHO, this is a great idea!  Saying this, I would love you to vote :+1: or :-1: on whether to record all our meetings notes going forward.  Thanks!

> Note, at this point, we're shifting the thread to this GitHub discussion.  As of 2024-12-15 17:54, there were 7 👍 and 0 👎

**Reply by dennyglee** [reply]
> From @ognis1205 at 2024-12-14 08:02:53

[@Denny Lee](https://unitycatalog.slack.com/team/U076YRD6LB0)
 I understand that some people, especially those at Databricks, are actively working hard on development, and I have great respect for such efforts. I also realize that this is a busy time for everyone. However, I want you to be aware that members outside of Databricks are also eager to participate in the project. I believe that if communication were more active, or if people showed more interest in each other's opinions, the project could become even better. I understand that communication and governance in OSS are challenging issues though.

**Reply by dennyglee** [reply]
Hey @ognis1205, I completely agree and am very happy with folks who want to work on the project and glad to have those discussions.  Saying this, we had bi-weekly (and at times weekly) open community meetups and regular communications via Slack.  As noted previously, we will start creating and publishing the results of these meetings into GitHub discussions to help with the communication.  Are there other forums or methods that may also be helpful?

I did have some folks suggest that we start using our email threads but they have not been active.  We can start doing this again to see if this helps gain traction - perhaps using [unity-catalog-user-discuss](https://lists.lfaidata.foundation/g/unity-catalog-user-discuss).  WDYT?

**Reply by ognis1205** [reply]
Thank you for sharing your information and ideas @dennyglee. I think all of your suggestions—holding community meetups, documenting the minutes on GitHub, and utilizing the LFAIDATA mailing list—are great ideas (and I’m aware that Denny and the product team have already been making similar efforts). I would definitely like to see progress in that direction.

Personally, I don’t think the core issue lies there. Rather, I believe the problem is that even when such communication tools are used, there tends to be a lack of feedback. Specifically, I suspect that many people feel uncertain because there’s little to no response on GitHub issues or PRs. It might help if contributors had more opportunities to understand the maintainers’ perspectives on what kinds of issues, PRs, or contributors are valued in the UC project.

I understand that everyone is very busy right now, so I think it’s necessary to continue observing the situation a bit more. However, I also believe it’s essential to implement measures to reduce the maintainers’ workload, such as improving the codebase to ease the burden of reviews.

Hesitating to say this to someone like Denny, who has such a profound understanding of OSS, though.

**Reply by dennyglee** [reply]
Ah, I got it; I think that's a fair assertion, though I would say that it depends on which areas.  I think around AI, we're overly communicative, but for other areas (e.g., managed tables), we're not communicative enough.  We're working on fixing this as we're finally very close to getting the technical charter published (lots of fun legalese back and forth).   With this in place, it'll be easier to officially identify whom are the maintainers of the project and criteria for new maintainers so we can ensure there are a lot more folks ready and available to answer questions, eh?!

**Comment by dennyglee** [comment]
> From @ni-mi at 2024-12-10 11:34:57 AM

[@TD](https://unitycatalog.slack.com/team/U07877VNK0V)
 as this is an open source project - I think that decisions about scope changes, timeline changes etc. should be taken together with the community - and not within Databricks conference rooms - if that's not the case than this project is not open source project - it's a project that Databricks are doing and letting the public to see it's code...
While I know there are issues with the version - and some of them at least are due to the fact that Delta are doing some refactoring on coordinated commits - I can't find any discussion about that not here, not in github - and not even in Delta Lake slack...
[@TD](https://unitycatalog.slack.com/team/U07877VNK0V)
 
[@Denny Lee](https://unitycatalog.slack.com/team/U076YRD6LB0)
 
[@Michelle Leon](https://unitycatalog.slack.com/team/U076V4KRP39)
 
[@Matthew Powers](https://unitycatalog.slack.com/team/U0780MEALEN)
 and if I forgot anyone else please add them as well - I know I want to be part of the community and I'm sure many more around here want to contribute - and that can only happen if things such as this will be discussed in public with the community...

**Reply by dennyglee** [reply]
Hey @ni-mi ([@Nimrod Ofek](https://unitycatalog.slack.com/team/U079ZQVGZFA)) - I always appreciate your candor.  You are right that this is an open-source project.  In fact, the issues you're talking about involve multiple open-source projects - Unity Catalog, Delta, Spark, Trino, Flink, Rust, DataFusion, Iceberg, Hudi, Polaris, etc.  There are multiple designs around coordinate commits that involve many different projects and if you look at those projects, there probably isn't a lot of conversations there either.
The main reason is that we're still designing (and re-designing based on feedback from other contributors - and not just from UC) so we can document the ideas into a PRD for everyone to review.  If we prematurely provide this information, it will only cause more confusion, not less.  As we had noted in previous threads and Unity Catalog community meetups, this is why we were focusing on the AI library integrations because there was a clear path forward for this quarter.  This would then allow all the contributor discussions across multiple projects to proceed so we could create a PRD that would explain all the nuances.

**Reply by dennyglee** [reply]
> From @ni-mi at 2024-12-14 10:41:47 AM

[@Denny Lee](https://unitycatalog.slack.com/team/U076YRD6LB0)
 thanks for the reply.
I'll point to two things:
1. I understand that this issue effects many oss projects. I think it should have been raised a long time ago in the Delta community, so people will be able to participate in the discussion - so the PRD will be with more community involvement. Also - who should write such a PRD? What was the process of choosing someone to write it? As an open source project- I believe Delta should have let the community at least the chance to vote on the person to do it... But that's for Delta project...

2. Having said that - UC is not Delta, and as a different open source project that was effected by Delta- I think we should have had an internal discussion/ vote about should we wait for coordinated commits for the initial version of managed tables? Should we move managed tables from 0.3.0 further down the road to when Delta will stabilise? Should we postpone v0.3.0 to when Delta will be stabilise around coordinated commits? even if there is no clear idea what is going on with Delta - it should be treated like any other issue and we should decide on course of action as a community - this is not a decision for Databricks product - unless this is not an open source project, than we shouldn't waste our time because all is decided internally within Databricks

3. We want, and are actually eager to participate, as 
[@OKAWA Shingo](https://unitycatalog.slack.com/team/U078RM1RN8Y)
 mentioned. If all decisions and discussions are being done within Databricks - we won't be able to contribute, not even minor code changes- and I think that will be a shame to miss the community...

**Reply by dennyglee** [reply]
**Part 1 of 3** 

Hey @ni-mi, allow me to answer each of your questions in its own thread so that way we can bifurcate and dive into each issue:

> I understand that this issue effects many oss projects. I think it should have been raised a long time ago in the Delta community, so people will be able to participate in the discussion - so the PRD will be with more community involvement. Also - who should write such a PRD? What was the process of choosing someone to write it? As an open source project- I believe Delta should have let the community at least the chance to vote on the person to do it... But that's for Delta project...

There may be a disconnect here, so allow me to clarify.  Concerning the issues around managed tables, while the initial focus is around Apache Spark™ and Delta Lake, please note that my call out for managed tables is that its potential impact on multiple lakehouse formats, query engines, and languages.  Because it involves so many different systems, we're still at the research stage for this.  

I do agree that PRD should be a community involvement but if folks haven't discovered all of the issues yet, the best we can do is try to communicate the current state. As noted in previous threads, we tried (obviously not as well as we should have) to communicate (via biweekly - sometimes weekly - community meetups) that we're still doing the research and that this impacts the 0.3 release timeline.  But to ensure that we were launching forward, we were doing a lot around `unitycatalog-ai` as those designs were more straightforward. 

> As noted in this [thread](https://github.com/unitycatalog/unitycatalog/discussions/803#discussioncomment-11575868), would it help if we published the meetup notes to GitHub and/or started using the email DLs again (though they didn't seem to get any traction the first time we tried to use them).

In terms of who is choosing to write it, just like any other distributed OSS project, whoever is leading the charge around a particular feature will be the one(s) to write it. If you would like to participate, in general, contact the owner or thread just like you did. For some features, it will take time for folks to write the document due to their complexity.

**Reply by dennyglee** [reply]
**Part 2 of 3**

Hey @ni-mi, 

> Having said that - UC is not Delta, and as a different open source project that was effected by Delta- I think we should have had an internal discussion/ vote about should we wait for coordinated commits for the initial version of managed tables? Should we move managed tables from 0.3.0 further down the road to when Delta will stabilise? Should we postpone v0.3.0 to when Delta will be stabilise around coordinated commits? even if there is no clear idea what is going on with Delta - it should be treated like any other issue and we should decide on course of action as a community - this is not a decision for Databricks product - unless this is not an open source project, than we shouldn't waste our time because all is decided internally within Databricks

As noted in the [previous thread](https://github.com/unitycatalog/unitycatalog/discussions/803#discussioncomment-11575928), the discussion around managed tables is not limited to Delta. Sure, we can find some time to discuss coordinated commits and their potential impact on the schedule. As noted in earlier threads, we'd like to finish the research before the PRD gets written.  But once we have the research in decent shape, we can start a discussion even before we start writing the PRD. 

Please note, as noted earlier in these threads and previous discussions per bi-weekly/weekly community meetups, the issues around managed UC tables and coordinated commits are not a Databricks product decision.  There are many lakehouse formats, query engines, and languages involved.  i.e., there are various lengthy discussions with various maintainers and committers across multiple projects.

**Reply by dennyglee** [reply]
**Part 3 of 3**

Per @ni-mi : 
> We want, and are actually eager to participate, as
[@OKAWA Shingo](https://unitycatalog.slack.com/team/U078RM1RN8Y)
mentioned. If all decisions and discussions are being done within Databricks - we won't be able to contribute, not even minor code changes- and I think that will be a shame to miss the community...

I'm glad that you want to participate eagerly! As noted in previous discussions, the best way to participate is to contribute to Unity Catalog. We have had many contributions to the UC codebase, and we welcome any and all contributions, from advocating to documentation to code/internals.    

While the managed tables and coordinated commits are an important component of UC, please note this is not the only one.  There have been a lot of changes to the codebase since even our 0.2 release.  As noted in other threads:

1. Previous threads and community meetups discussed a lot of our focus around `unity catalog-ai.`  Also noted earlier, we will start publishing notes from the community meetups to help with the communications.  Please feel free to follow up on this [thread](https://github.com/unitycatalog/unitycatalog/discussions/803#discussioncomment-11575868) if there are better forums outside of weekly/biweekly meetups meeting notes and potentially the restarting of the UC discuss email distribution list.
2. Once the research is in a decent state, we will find some time to discuss this issue
3. We have many issues and PRs open, we'd love to have your feedback on these as well
